pkg/cli/verifier/envoy_config.go

```diff
@@ -0,0 +1,248 @@
+		return result
+	}
+
+	//
+	// Verify the config
+
+	// Check for outbound listener
+	listeners := config.Listeners.GetDynamicListeners()
+	var outboundListener xds_listener.Listener
+	for _, l := range listeners {
+		if l.Name == lds.OutboundListenerName {
+			active := l.GetActiveState()
+			if active == nil {
+				result.Status = Failure
+				result.Reason = fmt.Sprintf("Outbound listener %q on source pod %q is not active", lds.OutboundListenerName, v.configAttr.trafficAttr.SrcPod)
+				return result
+			}
+			//nolint: errcheck
+			//#nosec G104: Errors unhandled
+			active.Listener.UnmarshalTo(&outboundListener)
+			break
+		}
+	}
+
+	if outboundListener.Name != lds.OutboundListenerName {
+		result.Status = Failure
+		result.Reason = fmt.Sprintf("Outbound listener %q not found on source pod %q", lds.OutboundListenerName, v.configAttr.trafficAttr.SrcPod)
+		return result
+	}
+
+	// Next, if the destination service is known, verify it has a matching filter chain
+	if v.configAttr.trafficAttr.DstService != nil {
+		dst := v.configAttr.trafficAttr.DstService
+		svc, err := v.kubeClient.CoreV1().Services(dst.Namespace).Get(context.Background(), dst.Name, metav1.GetOptions{})
+		if err != nil {
+			result.Status = Failure
+			result.Reason = fmt.Sprintf("Destination service %q not found: %s", dst, err)
+			return result
+		}
+		if err := v.findOutboundFilterChainForService(svc, outboundListener.FilterChains); err != nil {
+			result.Status = Failure
+			result.Reason = fmt.Sprintf("Did not find matching outbound filter chain for service %q: %s", dst, err)
+			return result
+		}
+	}
+
+	result.Status = Success
+	return result
+}
+
+func (v *EnvoyConfigVerifier) findOutboundFilterChainForService(svc *corev1.Service, filterChains []*xds_listener.FilterChain) error {
+	if svc == nil {
+		return nil
+	}
+
+	svcNamespacedName := fmt.Sprintf("%s/%s", svc.Namespace, svc.Name)
+
+	// There should be a filter chain for each port on the service.
+	// Each of those filter chains should match the clusterIP if set, otherwise
+	// to a list of pod IP ranges backing the service
+	dstIPRanges := mapset.NewSet()
+	if len(svc.Spec.ClusterIP) == 0 || svc.Spec.ClusterIP == corev1.ClusterIPNone {
+		endpoints, err := v.kubeClient.CoreV1().Endpoints(svc.Namespace).Get(context.Background(), svc.Name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Errorf("Endpoints not found for service %q", svcNamespacedName)
+		}
+		for _, sub := range endpoints.Subsets {
+			for _, ip := range sub.Addresses {
+				dstIPRanges.Add(ip.IP)
+			}
+		}
+	} else {
+		dstIPRanges.Add(svc.Spec.ClusterIP)
+	}
+
+	for _, port := range svc.Spec.Ports {
+		meshSvc := service.MeshService{
+			Name:      svc.Name,
+			Namespace: svc.Namespace,
+			Protocol:  v.configAttr.trafficAttr.AppProtocol,
+			Port:      uint16(port.Port),
+		}
+		if err := findOutboundFilterChainForServicePort(meshSvc, dstIPRanges, filterChains); err != nil {
```

Comment:
Should we check all ports before returning an error?

Reply:
This keeps it simple enough that we don't need to process a list of errors for every error path. Running the tool again after fixing the error seems low overhead. The general approach I have taken in this tool is to terminate at the first error, as typically an error could result in a cascade of errors that all relate to the root error.